### ISSUE TITLE
Connection property zeroDateTimeBehavior rejects value convertToNull when URL contains aws:

### ISSUE THREAD

**Issue by pukkaone** [awslabs/aws-mysql-jdbc#411]
### Describe the bug

Since [MySQL Connector/J 8.0.13](https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-13.html), Connector/J translates the legacy value of `convertToNull` for the connection property `zeroDateTimeBehavior` to `CONVERT_TO_NULL`. AWS JDBC Driver for MySQL also does this translation, but not when the URL contains `aws:`.

### Expected Behavior

Accept the legacy value `convertToNull` for the connection property `zeroDateTimeBehavior`.

### Current Behavior

On an attempt to connect to the example URL:
```
jdbc:mysql:aws://DUMMY-HOST-DOES-NOT-EXIST:3306/database?zeroDateTimeBehavior=convertToNull
```
the driver throws the exception:
```
Caused by: java.sql.SQLException: The connection property 'zeroDateTimeBehavior' acceptable values are: 'CONVERT_TO_NULL', 'EXCEPTION' or 'ROUND'. The value 'convertToNull' is not acceptable.
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:73)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:85)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.initSettings(ConnectionProxy.java:230)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.<init>(ConnectionProxy.java:105)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.<init>(ConnectionProxy.java:92)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.<init>(ConnectionProxy.java:80)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.autodetectClusterAndCreateProxyInstance(ConnectionProxy.java:128)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:221)
	at com.auction.jdbc.DriverIT.lambda$when_single_connection_aws_then_zeroDateTimeBehavior_convertToNull_accepted$1(DriverIT.java:24)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:55)
	... 72 more
Caused by: software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.WrongArgumentException: The connection property 'zeroDateTimeBehavior' acceptable values are: 'CONVERT_TO_NULL', 'EXCEPTION' or 'ROUND'. The value 'convertToNull' is not acceptable.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.DefaultPropertySet.initializeProperties(DefaultPropertySet.java:202)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.ha.ConnectionProxy.initSettings(ConnectionProxy.java:228)
	... 79 more
Caused by: software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.CJException: The connection property 'zeroDateTimeBehavior' acceptable values are: 'CONVERT_TO_NULL', 'EXCEPTION' or 'ROUND'. The value 'convertToNull' is not acceptable.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:151)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:124)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.EnumPropertyDefinition.parseObject(EnumPropertyDefinition.java:64)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.EnumPropertyDefinition.parseObject(EnumPropertyDefinition.java:39)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.AbstractRuntimeProperty.setValueInternal(AbstractRuntimeProperty.java:184)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.AbstractRuntimeProperty.initializeFrom(AbstractRuntimeProperty.java:81)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.DefaultPropertySet.initializeProperties(DefaultPropertySet.java:199)
	... 80 more
Caused by: java.lang.IllegalArgumentException: No enum constant software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.PropertyDefinitions.ZeroDatetimeBehavior.CONVERTTONULL
	at java.base/java.lang.Enum.valueOf(Enum.java:273)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.conf.EnumPropertyDefinition.parseObject(EnumPropertyDefinition.java:62)
	... 84 more
```

### Reproduction Steps

In the following unit test method, the URL does not contain `aws:`. The unit test method succeeds, and by succeeding, demonstrates the driver accepts the connection property `zeroDateTimeBehavior=convertToNull`.
```java
@Test
void when_single_connection_then_zeroDateTimeBehavior_convertToNull_accepted() throws Exception {
    var url = "jdbc:mysql://DUMMY-HOST-DOES-NOT-EXIST:3306/database?zeroDateTimeBehavior=convertToNull";
    var driver = new software.aws.rds.jdbc.mysql.Driver();
    assertThrows(CommunicationsException.class, () ->
            driver.connect(url, null));
}
```
In the following unit test method, the URL contains `aws:`. The unit test method fails, and by failing, demonstrates the driver rejects the connection property `zeroDateTimeBehavior=convertToNull`.
```java
@Test
void when_single_connection_aws_then_zeroDateTimeBehavior_convertToNull_accepted() throws Exception {
    var url = "jdbc:mysql:aws://DUMMY-HOST-DOES-NOT-EXIST:3306/database?zeroDateTimeBehavior=convertToNull";
    var driver = new software.aws.rds.jdbc.mysql.Driver();
    assertThrows(CommunicationsException.class, () ->
            driver.connect(url, null));
}
```

### The AWS JDBC Driver for MySQL version used

1.1.7

### JDK version used

openjdk version "17.0.7" 2023-04-18 OpenJDK Runtime Environment Homebrew (build 17.0.7+0) OpenJDK 64-Bit Server VM Homebrew (build 17.0.7+0, mixed mode, sharing)

### Operating System and version

MacOS 11.7.7 (Intel)

**Comment by crystall-bitquill** [comment]
Hi @pukkaone,

Thanks for reaching out and raising this issue.

We'll take a look at this and keep you updated as we investigate.

Thank you for your patience!

**Comment by crystall-bitquill** [comment]
Hi @pukkaone,

We've just merged a fix (#413). Could you kindly checkout our [snapshot build](https://github.com/awslabs/aws-mysql-jdbc#using-a-snapshot-of-the-driver) and let us know if the issue still persists?

Thank you!

**Comment by pukkaone** [comment]
I downloaded aws-mysql-jdbc-1.1.8-20230615.180623-6.jar and it works. Thanks.
